### ISSUE TITLE
Continuous dropout layer

### ISSUE THREAD

**Issue by link-er** [pytorch/pytorch#95290]
### 🚀 The feature, motivation and pitch

Hello!

I am working on information theory application for neural networks [(here)](https://openreview.net/forum?id=bQB6qozaBw).
With my research I show that for valid mutual information measurements one might apply continuous dropout. While continuous dropout was considered already in the original paper introducing dropout, the implementation of it is not unified and not added to the library. From my perspective it can be a large benefit to add a class for Gaussian dropout for example, or maybe for a dropout with noise sampled from any custom distribution.

Since I implemented it for my project, I am willing to create this contribution if it is considered interesting.

**Comment by jbschlosser** [comment]
Hey @link-er, thanks for your willingness to contribute! In general, to keep the maintenance burden manageable, note that our policy within torch.nn is to only accept new modules once they've been widely adopted, and then only if there are clear performance-related benefits or common expectations from the community that they be provided.

I see that the paper you've linked is fairly recent, and an older paper introducing continuous dropout ([arXiv](https://arxiv.org/abs/1911.12675), [PapersWithCode](https://paperswithcode.com/paper/continuous-dropout)) has 66 citations. Given this, the criteria for inclusion within torch.nn are not met at this time, but we will happily reconsider in the future if the technique increases in popularity. We can leave the issue open to gauge popularity among other users.

**Comment by link-er** [comment]
Hey! Thanks for the answer, makes total sense to me.

I am sorry for the confusion in terminology, but the paper you linked (Continuous dropout) is not exactly what I am proposing to add. This paper proposes to sample a layer right away from some distribution and they term it continuous dropout.

On the other hand, what I am talking about is simply replacing the noise distribution from Bernoulli to Gaussian. If you look into the original dropout paper https://jmlr.org/papers/volume15/srivastava14a/srivastava14a.pdf Section10 (and Table10), then you see why I think it might be beneficial. 

I cannot tell for sure, why Gaussian noise was never widely used for dropout, maybe because people were satisfied with performance of Bernoulli enough.

**Comment by link-er** [comment]
p.s. Sorry, my fault, I mixed up the paper with FastDroput paper =)

You are indeed correct that this one is also about continuous dropout, but still - it was not really proposed by them.
